src/Update.py

```diff
@@ -41,7 +41,8 @@ def run_migrations():
                 ans = Run.prg("python", migration.absolute(), useSubprocess=True)
 
                 if ans == 0:
-                    Slack.upload_log()
+                    Discord().post_log(Log.dump())
```

Comment:
Does [the addition of the `Discord.notify()` class method](https://github.com/vindennl48/LOFSongManager/pull/10/commits/cd8798947b47eb06f52d2d5ce1fbae08b2c6270a) resolve this?

Reply:
I pulled your comment from the other comment since it essentially is about this.
> James -
> This one I do have strong feelings about. The thing I really want to avoid is having implicit behavior in the constructor. In my opinion, the constructor shouldn't really do anything other than return an instance of the class. My understanding of good object-orientated design is that an object's behavior should be defined in its methods. In my experience, this tends to work out well over the long-term, and the benefits outweigh the drawbacks (namely, more verbose and ugly code). I can definitely elaborate on some of the specific benefits if you're not fully sold on this.

> All of that said, I think we can still improve this. Take a look at [this commit](https://github.com/vindennl48/LOFSongManager/pull/10/commits/cd8798947b47eb06f52d2d5ce1fbae08b2c6270a). Instead of having to instantiate a new object with Discord() and then chaining a post_* method every time, I thought it would be cleaner to have a single class method that would accept a dict of options and data. The result is that notifications are created by calling Discord.notify({ ... }).

> It's still not perfect, but I think it makes for a better interface, and it feels more natural than how it was before. What do you think?

I believe the disconnect here is that I'm using the class as a static class, basically imitating how C++ works with Namespaces.  Because they do not need an instance, they should be global throughout the project or else we will get all sorts of circular dependency issues.  The other reason I had done it this way (I did the same for the Log class as well) was to make things easier to read and remember how things worked.  Instead of having to memorize the member functions to make something happen, it is built inherently into the 'constructor' (loosely used term here because of how I am using the class).  It was a conscious decision to build it that way at the time.

I'm not too keen on refactoring the whole project again when it isn't necessarily needed.  I would like everything the same so if this class changes, then they all should be changed.